### PR TITLE
New: Floating Bars - Variation 1

### PR DESCRIPTION
**Pull request by mserino**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/floating-bars/widgets/floating-bars-var-1.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	public function get_name(): string {
+		return 'floating-bars-var-1';
+	}
+
+	public function get_title(): string {
+		return esc_html__( 'Floating Bar CTA', 'elementor' );
+	}
+
+    public function get_group_name(): string {
```
Comment (mykytamurzin) [inline modules/floating-bars/widgets/floating-bars-var-1.php:28]: something weird with indenting

File: modules/floating-bars/classes/render/floating-bars-core-render.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+		$this->widget->add_render_attribute( 'cta-button', [
+			'class' => $cta_classnames,
+		] );
+
+		if ( ! empty( $link['url'] ) ) {
+			$this->widget->add_link_attributes( 'cta-button', $link );
+			?>
+				<div class="e-floating-bars__cta-button-container">
+					<a <?php echo $this->widget->get_render_attribute_string( 'cta-button' ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
```
Comment (mykytamurzin) [inline modules/floating-bars/classes/render/floating-bars-core-render.php:80]: can we add sanitization here somehow? at least for href attribute to protect from cases like 
`<a href="javascript:alert('Hello, world!');">Click me</a>`

Comment (mserino) [reply modules/floating-bars/classes/render/floating-bars-core-render.php:80]: Hi @mykytamurzin do you have a suggestion on the best approach to sanitize this? I got this code from the elementor docs: https://developers.elementor.com/docs/editor-controls/control-url/

Comment (mykytamurzin) [reply modules/floating-bars/classes/render/floating-bars-core-render.php:80]: You can try to use `esc_url` here
```
$link = esc_url( $this->settings['cta_link'] ) ?? '';
```
or 
```
$this->widget->add_link_attributes( 'cta-button', esc_url( $link ) );
```

Comment (mserino) [reply modules/floating-bars/classes/render/floating-bars-core-render.php:80]: @mykytamurzin I don't think this is necessary. Looking at the elementor code, the `add_link_attributes` function already escapes the url:

`$attributes['href'] = esc_url( $url_control['url'], $allowed_protocols );`
(this comes from `includes/base/element-base.php` line 413)

Comment (mykytamurzin) [reply modules/floating-bars/classes/render/floating-bars-core-render.php]: good then, thanks for checking

File: modules/floating-bars/widgets/floating-bars-var-1.php (hunk 1 above)
Comment (mserino) [reply modules/floating-bars/widgets/floating-bars-var-1.php]: fixed

**Review by murzin-nikita — DISMISSED** [review]

**Review by murzin-nikita — APPROVED** [review]

**Pull request merged**